### ISSUE TITLE
Can we have the spark version upgrade to 3.3.0 . The version used right now has log4j vulnerabilities.

### ISSUE THREAD

**Issue by psb2509** [AbsaOSS/cobrix#518]
## Background
In each of the pom file used in cobrix we have older verison of spark which has log4j vulnerabilities . 

## Feature
Can we have the version updated to 3.3.0 for spark throughout the application and new version of cobrix released in maven. 

## Example [Optional]
[A simple example if applicable.](https://github.com/itspsb/cobrix)

## Proposed Solution [Optional]
Solution Ideas 
1. https://github.com/itspsb/cobrix
2. 
3.

**Comment by yruslan** [comment]
Yes, you can build for Spark 3.3.0. See the section on building the uber jar:
https://github.com/AbsaOSS/cobrix#creating-an-uber-jar

**Comment by itspsb** [comment]
I created the uber jar and installed it on databricks cluster . 
![image](https://user-images.githubusercontent.com/114199586/192294924-33a03ac7-6fab-485d-84c5-f31d7091ecdf.png)
When i run an example i get this error 
![image](https://user-images.githubusercontent.com/114199586/192295068-9d73d1d6-29a5-4ae3-8d38-dfcee6c6e4b5.png)

Can you suggest what i might be doing wrong here . All i needed is a jar file like this on mavencentral. 
https://mvnrepository.com/artifact/za.co.absa.cobrix/spark-cobol_2.12/2.5.1
https://repo1.maven.org/maven2/za/co/absa/cobrix/spark-cobol_2.12/2.5.1/spark-cobol_2.12-2.5.1.jar

**Comment by yruslan** [comment]
Looks like a Scala version mismatch. Which Scala version is used for the Databricks cluster?

**Comment by yruslan** [comment]
For Databricks clusters creating a custom Uber jar is recommended. Maven Central contains only thin jars of Cobrix. This is because it is up to Maven to handle dependencies on the build. An uber jar created by 'sbt assembly' contains 'spark-cobol' and all necessary dependencies to use it on a Spark cluster such as Databricks.

**Comment by itspsb** [comment]
Does this means i simple run the jar:jar plugin for spark cobol in intelij ide ?
Do you have any steps for creating the custom uber jar for databricks ?

**Comment by itspsb** [comment]
Does this means i simple run the jar:jar plugin for spark cobol in intelij ide ?
Do you have any steps for creating the custom uber jar for databricks ?

**Comment by yruslan** [comment]
Scala 2.13 is not supported yet. It is a work in progress and is going to be released in 2.6.0.
You can build the uber jar for Scala 2.13 from master.

As I've pointed out, steps to create an uber jar are specified here: https://github.com/AbsaOSS/cobrix#creating-an-uber-jar

In your case the command will be:
```
sbt ++2.13.8 assembly -DSPARK_VERSION=3.2.2
```
You can collect the assembly jar at `spark-cobol/target/scala-2.13/spark-cobol-assembly-2.5.2-SNAPSHOT.jar`
This is the JAR you can upload to Databricks.

**Comment by itspsb** [comment]
Here is the steps I followed . 
I forked the master branch into this repo `https://github.com/itspsb/cobrix` . 
Made the changes to pom files to point to update the version for spark to spark 3.3.0 and then ran 
`sbt ++2.13.8 assembly -DSPARK_VERSION=3.3.0` . I am assuming since i am using new spark version i should use spark 3.3.0 in the command . Is is not correct ?

**Comment by yruslan** [comment]
Yes, looks correct. 

No need to change anything in POM files though. Sbt build uses 'build.sbt'

**Comment by psb2509** [comment]
is there a particular version of java needed to compile the code here . Does this work with java 11 as well ?

**Comment by yruslan** [comment]
It compiles only with Java 8

**Comment by itspsb** [comment]
Thank You @yruslan . I was able to fix all the issues and upgrade to version 3.3.0 . Had to delete the example folderand hard code  path to jar in the pom file for cobol parser to make sure it wasnt pointing to the  cobol parser from mavencentral . I then uploaded all 3 jars to databricks cluster and then it worked like a charm . Thanks a lot for building this awesome library !! .
